wgpu-native/src/command/mod.rs

```diff
@@ -249,20 +249,42 @@ pub fn command_encoder_begin_render_pass<B: GfxBackend>(
             "Attachment sample_count must be supported by physical device limits"
         );
         for at in color_attachments.iter() {
-            let sample_count_check = view_guard[at.attachment].samples;
+            let color_attachment_view = &view_guard[at.attachment];
             assert_eq!(
-                sample_count_check, sample_count,
+                color_attachment_view.samples, sample_count,
                 "All attachments must have the same sample_count"
             );
 
-            if let Some(resolve) = unsafe { at.resolve_target.as_ref() } {
+            let color_attachment_texture = &texture_guard[color_attachment_view.texture_id.value];
+            assert!(
+                color_attachment_texture.usage.contains(TextureUsage::OUTPUT_ATTACHMENT),
+                "Color attachment texture must be created with TextureUsage::OUTPUT_ATTACHMENT",
```

Comment:
Given the proper variable names, the assertion is pretty self-describing through the code. The comment seems to be just repeating the same thing, so it doesn't look necessary.

Reply:
👍  I didn't realize `assert!` actually gives a pretty decent error message (unlike `assert_eq!`)